libs/hdf-converters/src/converters-from-hdf/splunk/reverse-splunk-mapper.ts

```diff
@@ -432,48 +431,64 @@ export class FromHDFToSplunkMapper extends FromAnyBaseConverter {
     }
   }
 
-  async toSplunk(
-    config: SplunkConfig,
-    filename: string,
-    webCompatibility = false
-  ): Promise<string> {
-    let service: splunkjs.Service;
-    if (webCompatibility) {
-      service = new splunkjs.Service(new ProxyHTTP.JQueryHttp(''), config);
-    } else {
-      service = new splunkjs.Service(config);
-    }
+  async toSplunk(config: SplunkConfig, filename: string): Promise<string> {
+    const hostname = config.port
+      ? `${config.scheme}://${config.host}:${config.port}`
+      : `${config.scheme}://${config.host}:8089`;
     logger.info(
-      `Logging into Splunk Service: ${config.host} with user ${config.username}`
+      `Logging into Splunk instance at ${hostname} with user ${config.username}`
     );
-    logger.verbose('Got Execution: ' + filename);
+
+    logger.verbose(`Found designated file to transfer: ${filename}`);
     const guid = createGUID(30);
-    logger.verbose('Using GUID: ' + guid);
-    return new Promise((resolve) => {
-      if (config.username && config.password) {
-        service.login((error: any) => {
-          this.handleSplunkError(error);
-        });
+    logger.verbose(`Using GUID: ${guid}`);
+
+    // Attempt to authenticate using given credentials
+    const username = (config.username ??= '');
+    const password = (config.password ??= '');
+    const authResponse = await this.axiosInstance.post(
+      `${hostname}/services/auth/login`,
+      new URLSearchParams({
+        username: username,
+        password: password
+      })
+    );
+    this.axiosInstance.defaults.headers.common[
+      'Authorization'
+    ] = `Bearer ${authResponse.data.sessionKey}`;
+
+    // Request all available indexes
+    const indexResponse = await this.axiosInstance.get(
+      `${hostname}/services/data/indexes`,
+      {
+        params: {count: 0}
       }
-      service.indexes().fetch(async (error: any, indexes: any) => {
-        if (error) {
-          this.handleSplunkError(error);
-        } else if (!indexes) {
+    );
+
+    return new Promise((resolve) => {
```

Comment:
when using promise + resolve/reject syntax you need to always make sure to handle the potential error case.  ex. throwing/catching errors like this definitely isn't gonna work out how you want it to

Reply:
Changed as requested.